pkg/limits/store.go

```diff
@@ -295,19 +318,45 @@ func (s *usageStore) getStripe(tenant string) int {
 	return int(h.Sum32() % uint32(len(s.locks)))
 }
 
+// getPartitionForHash returns the partition for the hash.
+func (s *usageStore) getPartitionForHash(hash uint64) int32 {
+	return int32(hash % uint64(s.numPartitions))
+}
+
+// withinActiveWindow returns true if t is within the active window.
+func (s *usageStore) withinActiveWindow(t time.Time) bool {
```

Comment:
I will update the code to use this function instead of copy/pasting the `cutoff` calculation in a subsequent PR.